consensus/hotstuff/timeoutcollector/timeout_processor.go

```diff
@@ -243,17 +245,36 @@ func (p *TimeoutProcessor) validateTimeout(timeout *model.TimeoutObject) error {
 // weight for building TC. This function is run only once by single worker.
 // Any error should be treated as exception.
 func (p *TimeoutProcessor) buildTC() (*flow.TimeoutCertificate, error) {
-	signers, highQCViews, aggregatedSig, err := p.sigAggregator.Aggregate()
+	signersData, aggregatedSig, err := p.sigAggregator.Aggregate()
 	if err != nil {
 		return nil, fmt.Errorf("could not aggregate multi message signature: %w", err)
 	}
 
+	// IMPORTANT: To properly verify an aggregated signature included in TC we need to provide list of signers with corresponding
+	// messages(`TimeoutCertificate.NewestQCViews`) for each signer, this relation should be very strict.
```

Comment:
```suggestion
	// messages(`TimeoutCertificate.NewestQCViews`) for each signer. If the one-to-once correspondence of view and signer is not
	// maintained, the TC's aggregated signature will be invalid. 
```

Reply:
> the TC's aggregated signature will be invalid.

An aggregated signature is never invalid on its own. It all depends on how the verification is performed : It is valid if it is verified against the correct list of (pub_key, message) that initially generated the signature, and it is invalid if it is verified against a different list. 
I would suggest a slightly different comment : 
"If the one-to-once correspondence of view and signer is not maintained, it won't be possible to verify the aggregated signature"